src/metadata_parser.py

```diff
@@ -409,3 +409,59 @@ def create_controlled_identified_term_value(
         }
 
         return nmdc_controlled_term_slot
+
+    def generate_csv(self):
```

Comment:
generate_example_biosample_csv ?  

I think we should have an output parameter, your default of "example_biosample_metadata.csv" is good though.